packages/rainbowkit/src/config/getDefaultConfig.ts

```diff
@@ -1,15 +1,39 @@
-import { type Chain, configureChains, createConfig } from 'wagmi';
-import { publicProvider } from 'wagmi/providers/public';
-import type { WalletList } from '../wallets/Wallet';
+import { Transport } from 'viem';
+import {
+  CreateConfigParameters,
+  WagmiProviderProps,
+  createConfig,
+} from 'wagmi';
+import type { Chain } from 'wagmi/chains';
+import type {
+  RainbowKitWalletConnectParameters,
+  WalletList,
+} from '../wallets/Wallet';
 import { connectorsForWallets } from '../wallets/connectorsForWallets';
-import type { WalletConnectConnectorMetadata } from '../wallets/getWalletConnectConnector';
 import {
   coinbaseWallet,
   metaMaskWallet,
   rainbowWallet,
   walletConnectWallet,
 } from '../wallets/walletConnectors';
 
+type _chains = readonly [Chain, ...Chain[]];
+
+// Define the '_transports' type as a Record
+// It maps each 'Chain' id to a 'Transport'
+type _transports = Record<_chains[number]['id'], Transport>;
+
+interface GetDefaultConfigParameters {
+  appName: string;
+  appDescription?: string;
+  appUrl?: string;
+  appIcon?: string;
+  chains: readonly [Chain, ...Chain[]];
+  wallets?: WalletList;
+  projectId: string;
+  transports: _transports;
```

Comment:
`transports` can be optional, and we just infer an `http()` map with the public provider defaults